src/coredata.cs

```diff
@@ -126,58 +147,74 @@ public interface NSAttributeDescription {
 	public interface NSEntityDescription : NSCoding, NSCopying {
 
 		[Static, Export ("entityForName:inManagedObjectContext:")]
+		[return: NullAllowed]
 		NSEntityDescription EntityForName (string entityName, NSManagedObjectContext context);
 
 		[Static, Export ("insertNewObjectForEntityForName:inManagedObjectContext:")]
-		NSObject InsertNewObjectForEntityForName (string entityName, NSManagedObjectContext context);
+		NSManagedObject InsertNewObjectForEntityForName (string entityName, NSManagedObjectContext context);
 
 		[Export ("managedObjectModel")]
 		NSManagedObjectModel ManagedObjectModel { get; }
 
 		[Export ("managedObjectClassName")]
 		string ManagedObjectClassName { get; set; }
 
-		[Export ("renamingIdentifier")]
+		[NullAllowed, Export ("renamingIdentifier")]
 		string RenamingIdentifier { get; set; }
 
 		[NullAllowed] // by default this property is null
 		[Export ("name")]
 		string Name { get; set; }
 
-		[Export ("Abstract")]
+		[Export ("abstract")]
```

Comment:
good catch, that was *not* an issue because `isAbstract` took precedence (but a good fix to have anyway)